src/main/java/duke/data/Storage.java

```diff
@@ -0,0 +1,119 @@
+package duke.data;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+/**
+ * Storage class to load tasks from file and save tasks in the file.
+ */
+public class Storage {
+    public static final char LINE_SEPARATOR = '|';
+    private File file;
+
+    /**
+     * Initializes Storage object with the filePath
+     *
+     * @param filePath Path to local file
+     */
+    public Storage(String filePath) {
+        this.setup(filePath);
+    }
+
+    /**
+     * Creates the data folder and file locally if it does not exist.
+     *
+     * @param filePath
+     */
+    public void setup(String filePath) {
+        String curDirectory = System.getProperty("user.dir");
+        String directoryName = "data";
+
+        File directory = new File(curDirectory + "/" + directoryName);
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+
+        File file = new File(curDirectory + "/" + filePath);
+        try {
+            if (!file.exists()) {
+                file.createNewFile();
+            }
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+        this.file = file;
+    }
+
+    /**
+     * Loads tasks from the local disk storage
+     *
+     * @return ArrayList of tasks to be used to initialize a TaskList object
+     */
+    public List<Task> load() {
+        List<Task> newList = new ArrayList<>();
+        try {
+            Scanner s = new Scanner(file);
```

Comment:
Perhaps you may want to make the Scanner name slightly longer than just "s" as it can be slightly confusing since the method also deals heavily with Strings?